tests/test_walker.py

```diff
@@ -166,8 +166,77 @@ def test_auto_template_identification():
     this_caldb.remove_entry(flat)
     this_caldb.remove_entry(bp_map)
 
-if __name__ == "__main__":
-    test_autoreducing()
+
+def test_saving():
+    """
+    Tests the special save function including suffix. Tries both calibration image and non-calibration image
+    """
+    ### create dirs
+    datadir = os.path.join(os.path.dirname(__file__), "simdata")
+    if not os.path.exists(datadir):
+        os.mkdir(datadir)
+    outputdir = os.path.join(os.path.dirname(__file__), "walker_output")
+    if not os.path.exists(outputdir):
+        os.mkdir(outputdir)
+
+    ### load test recipe
+    testdatadir = os.path.join(os.path.dirname(__file__), "test_data")
+    save_recipe_path = os.path.join(testdatadir, "saving_only.json")
+    save_recipe = json.load(open(save_recipe_path, 'r'))
+    
+    ######################
+    ## Test regular images
+    ######################
+
+    ### Create mock Image data
+    l1_dataset = mocks.create_dark_calib_files(filedir=datadir, numfiles=2)
+    # fake the emgain
+    for image in l1_dataset:
+        image.ext_hdr['EMGAIN'] = 1
```

Comment:
I feel like I've seen this a few places. Should we add it into the default headers we use for mock data?